cmd/auth-handler.go

```diff
@@ -125,29 +145,47 @@ func checkAdminRequestAuthType(r *http.Request, region string) APIErrorCode {
 	return s3Err
 }
 
-func checkRequestAuthType(ctx context.Context, r *http.Request, action policy.Action, bucketName, objectName string) APIErrorCode {
-	isOwner := true
-	accountName := globalServerConfig.GetCredential().AccessKey
+// Fetch the security token set by the client.
+func getSessionToken(r *http.Request) (token string) {
+	token = r.Header.Get("X-Amz-Security-Token")
+	if token != "" {
+		return token
+	}
+	token = r.URL.Query().Get("X-Amz-Security-Token")
+	if token != "" {
+		return token
+	}
+	return ""
```

Comment:
the above 4 lines can be replaced by
`return r.URL.Query().Get("X-Amz-Security-Token")`

Reply:
done